### ISSUE TITLE
FixColumnPartitions always called for RCB partitioning

### ISSUE THREAD

**Issue by raovgarimella** [MeshToolkit/MSTK#65]
Whenever RCB partitioning is called, it is assumed that the mesh has columns of elements (like in geological meshes) and FixColumnPartitions is called. However, if the mesh is not columnar, FixColumnPartitions can die with a FATAL error.

**Comment by raovgarimella** [comment]
We can do one of two things to resolve this:
1. If FixColumnPartitions cannot detect columns in the mesh, it can print out a non-fatal message and return
2. We can create a new partitioner called RCB_Columns which calls the Zoltan RCB partitioner and calls FixColumnPartitions. Calling plain RCB will not invoke this routine

**Comment by lipnikov** [comment]
FYI: I successfully called RCB partitioning for triangular meshes on non-manifolds..

**Comment by raovgarimella** [comment]
It will work on surface meshes because FixColumnPartitions does nothing in that case

**Comment by ekikinzon** [comment]
The ideal solution would be to have another parameter to enable/disable FixColumns. In fact, is there any reason why it's never used with other partitioners? I get an impression that it's exclusive for RCB only because in the context of Amanzi/ATS RCB was only accessible through meshconvert and no one was using meshconvert on fully unstructured grids with RCB option.

The two options proposed are definitely easier to implement as they require minimal changes. The issue with the first one is that if RCB is used from Amanzi, running FixColumns  might be undesirable even if it doesn't crash the code.
The second option basically requires to copy-paste the entire c file with the only difference being a single if statement, which wouldn't look pretty, but is probably better than the first option.

About surface meshes: FixColumns is only triggered when the number of regions is nonzero, i.e. we have a volume mesh (manifolds still have zero number of regions).

**Comment by raovgarimella** [comment]
------
The second option basically requires to copy-paste the entire c file with the only difference being a single if statement, which wouldn't look pretty, but is probably better than the first option.
------
No, we can call RCB partitioning for any partitioner that has RCB in it but call FixColumnPartitions only if the option is RCB_Columns

**Comment by lipnikov** [comment]
When copy-paste looks like a possible solution, use a factory.

**Comment by ekikinzon** [comment]
Fair enough, looking at the code FixColumns is the last thing that might touch the part array, so it can be applied after the partitioner is called. Which also means it can be applied to graph or even metis. 

The factory-like thing lives in MESH_Get_Partitioning.c, so we can throw in more options into 
switch (method)
with new ones calling FixColumns after MESH_PartitionWith* is called.

**Comment by ekikinzon** [comment]
Made a pull request, partition methods without FixColumns are now 0-2, with FixColumns are 3-5.
